clientpool/errors.go

```diff
@@ -1,12 +1,22 @@
 package clientpool
 
 import (
-	"errors"
 	"fmt"
 )
 
 // ErrExhausted is the error returned by Get when the pool is exhausted.
-var ErrExhausted = errors.New("clientpool: exhausted")
+var ErrExhausted = exhaustedError{}
+
+type exhaustedError struct{}
+
+func (exhaustedError) Error() string {
+	return "clientpool: exhausted"
+}
+
+func (exhaustedError) Retryable() *bool {
+	v := true
+	return &v
```

Comment:
We use `*bool` (over enum) so that packages implementing it do not need to import `retrybp` is they don't want to.

but if they want to import `retrybp`, then they can use `BoolPtr`.

Reply:
An alternative over `*bool` is to use `int`, with >0 means retryable, <0 means not, and =0 means no decision. This is similar to `strcmp` in c stdlib.

Reply:
> An alternative over *bool is to use int, with >0 means retryable, <0 means not, and =0 means no decision. This is similar to strcmp in c stdlib.

🤔  I kinda like that idea... @konradreiche what do you think?

Reply:
Same, I think it makes it more clear that there are three states as opposed to two :+1:

Reply:
ok changed to int.